### PR TITLE
http_proxy_errors: improve tls errors handling 

### LINKED ISSUE #549: verify that we return maximum available information when handling TLS errors
If there is a TLS error we cast it to TLS specific errors 

https://github.com/saucelabs/forwarder/blob/2bcdb6fe44168c021e2dcfc985567ed742645bf4/http_proxy_errors.go#L92

The TLS errors contain more information that we provide.
It would be super useful in debugging if we provided all the information like the certificate chain that failed etc.

### PR DESCRIPTION
**Pull request by Choraden**
Fixes #549

### CODE REVIEW

File: http_proxy_errors.go [hunk 1]
```diff
@@ -146,7 +146,7 @@ func handleNetError(req *http.Request, err error) (code int, msg, label string)
 }
 
 func handleTLSRecordHeader(req *http.Request, err error) (code int, msg, label string) {
-	var headerErr *tls.RecordHeaderError
+	var headerErr tls.RecordHeaderError
```
Comment (mmatczuk) [inline http_proxy_errors.go:149]: Looks like we could have some testing around it.

File: http_proxy_errors.go [hunk 2]
```diff
@@ -237,6 +242,14 @@ func handleStatusText(req *http.Request, err error) (code int, msg, label string
 	return
 }
 
+func tlsRecordHeaderLooksLikeHTTP(hdr [5]byte) bool {
+	switch string(hdr[:]) {
+	case "HTTP/", "GET /", "HEAD ", "POST ", "PUT /", "OPTIO":
```
[the remark this answers is not in the document]
Comment (Choraden) [reply http_proxy_errors.go:247]: Like this?
```go
func tlsRecordHeaderLooksLikeHTTP(hdr [5]byte) bool {
	return bytes.HasPrefix(hdr[:], []byte("HTTP/")) ||
		bytes.HasPrefix(hdr[:], []byte("GET /")) ||
		bytes.HasPrefix(hdr[:], []byte("HEAD ")) ||
		bytes.HasPrefix(hdr[:], []byte("POST ")) ||
		bytes.HasPrefix(hdr[:], []byte("PUT /")) ||
		bytes.HasPrefix(hdr[:], []byte("OPTIO"))
}
```
Looks worse imo.

[the remark this answers is not in the document]
Comment (mmatczuk) [reply http_proxy_errors.go:247]: my point is that it allows not to have `/`.

File: http_proxy_errors.go [hunk 3]
```diff
@@ -27,8 +27,12 @@ type denyError struct {
 	error
 }
 
-// ErrorHeader is the header that is set on error responses with the error message.
-const ErrorHeader = "X-Forwarder-Error"
+const (
```
[the remark this answers is not in the document]
Comment (Choraden) [reply http_proxy_errors.go]: ok

File: http_proxy_errors.go (hunk 1 above)
Comment (Choraden) [reply http_proxy_errors.go:147]: Added tests for record header and unverified cert.

File: http_proxy_errors.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Choraden) [reply http_proxy_errors.go]: OK. Done.

**Comment by mmatczuk** [comment]
Sweet.

**Review by mmatczuk — APPROVED** [review]

**Pull request merged**